eclipse-che/che-server
# feat: support retrieving Git user data when PAT is configured

### What does this PR do?
feat: support retrieving Git user data when PAT is configured

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-4057

### How to test this PR?

1. Deploy Eclipse Che `chectl server:deploy -p openshift --cheimage quay.io/abazko/che-server:crw-4057`
2. Create PAT for any OAuth provider
3. Check if `workspace-userdata-gitconfig-configmap` is created with proper data

### Tested
* Azure DevOps: PAT + OAuth
* Github (https://github.com): PAT + OAuth
* Gitalb (https://gitlab.com): PAT + OAuth
* Gitlab server: PAT + OAuth
* BitBucket server:  PATH + OAuth
* BitBucket: OAuth

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
